service/src/main/java/uk/gov/hmcts/reform/fpl/service/docmosis/GeneratedOrderTemplateDataGeneration.java

```diff
@@ -46,6 +48,12 @@
     public DocmosisGeneratedOrder getTemplateData(CaseData caseData) {
         OrderTypeAndDocument orderTypeAndDocument = caseData.getOrderTypeAndDocument();
         GeneratedOrderType orderType = orderTypeAndDocument.getType();
+        String dateOfIssue;
+        if (EMERGENCY_PROTECTION_ORDER.equals(orderType)) {
```

Comment:
does not look like correct place for things like this, it i generic class, should it not be overriden in EPOGenerationService?

Reply:
I agree but dateOfIssue is a generic field that is added at the end of builder in the abstract class - should I remove and add it to each __GenerationService?